garak/generators/rest.py

```diff
@@ -212,7 +212,16 @@ def _call_model(
             "proxies": self.proxies,
             "verify": self.verify_ssl,
         }
-        resp = self.http_function(self.uri, **req_kArgs)
+        try:
+            resp = self.http_function(self.uri, **req_kArgs)
+        except UnicodeEncodeError as uee:
+            # only RFC2616 (latin-1) is guaranteed
+            # don't print a repr, this might leak api keys
+            logging.error(
+                "Only latin-1 encoding supported by HTTP RFC 2616, check headers and values for unusual chars",
+                exc_info=uee,
+            )
+            raise BadGeneratorException from uee
```

Comment:
Having read the reported issue I think the raise here may not be an appropriate type. `BadGeneratorException` extends `PluginConfigurationError` and this error is not related to the configuration at this time.

This error seems like a new exception class may be appropriate, something like `BadResponseException` / `InvalidResponseError` / `MalformedResponseException`. The message helps the user understand that the issue is with the response data, however I am looking for something `programatic` to make it clear that the target systems response is not supported vs suggesting that configuration of the generator could be changed to avoid the issue.